### ISSUE TITLE
box: list available box.cfg options before box configuration

### ISSUE THREAD

**Issue by Totktonada** [tarantool/tarantool#8141]
Parameters to pass into `box.cfg()` may be received from an external configuration file/storage. In this case we should validate them before calling `box.cfg()`.

What else except the list of names is needed:

* Allowed types.
* Whether the option can be configured only once or dynamically at any time.

NB: It would be nice to replace the [hardcoded box.cfg options list](https://github.com/tarantool/cartridge/blob/56fe20ab8992d4cc283eb9d25d1498ce8f39e272/cartridge/argparse.lua#L112-L208) in cartridge to don't [update it constantly](https://github.com/tarantool/cartridge/pull/1826) and to don't let users update cartridge in order to use new box.cfg options.

**Comment by Totktonada** [comment]
It is duplicate of #3628. I moved all the metainfo to the linked issue. Closing this one.
